exporter/kinesisexporter/exporter.go

```diff
@@ -16,29 +16,41 @@ package kinesisexporter
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/google/uuid"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenterror"
 	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/consumer/pdata"
-	jaegertranslator "go.opentelemetry.io/collector/translator/trace/jaeger"
 	"go.uber.org/zap"
 )
 
+const (
+	errInvalidMarshaller = "invalid marshaller encoding"
+)
+
 // Exporter implements an OpenTelemetry exporter that pushes OpenTelemetry data to AWS Kinesis
 type Exporter struct {
-	producer producer
-	logger   *zap.Logger
+	producer   producer
+	logger     *zap.Logger
+	marshaller Marshaller
 }
 
 // newExporter creates a new exporter with the passed in configurations.
 // It starts the AWS session and setups the relevant connections.
 func newExporter(c *Config, logger *zap.Logger) (*Exporter, error) {
+	marshaller, valid := defaultMarshallers()[c.Encoding]
+	if !valid {
+		return nil, fmt.Errorf(errInvalidMarshaller)
```

Comment:
nit: would be helpful to mention the value for c.Encoding so the error reads like: `invalid marshaller encoding: "jaeger_thrift"`. Even better if error also mentioned valid choices but mentioning the user specified value would be good enough for me.